### PR TITLE
feat: sidebar v2

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Implement the new accordion sidebar

- Refactored sidebar config file by splitting into multiple sub-configs
- Modified `DocSidebarItem/Category` component to support displaying icons on the sidebar category.
- Added some placeholder doc pages

sidebar

### CODE REVIEW

File: src/sidebar/docs.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  //   label: 'User flows',
+  //   link: { type: 'doc', id: 'user-flows/README' },
+  //   items: [],
+  //   customProps: { id: 'user-flows' },
+  // },
+  // { customization: [], customProps: { id: 'customization' }, },
+  // { authorization: [], customProps: { id: 'authorization' }, },
+  // { customization: [], customProps: { id: 'customization' }, },
+  // { developer: [], customProps: { id: 'developer' }, },
+  // { elements: [], customProps: { id: 'elements' }, },
```
Comment (darcyYe) [inline src/sidebar/docs.ts:91]: need to remove or commented deliberately?

File: src/sidebar/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  tutorialSidebar: [
+    'intro',
+    'hello',
+    {
+      type: 'category',
+      label: 'Tutorial',
+      items: ['hello'],
+    },
+  ],
+   */
```
Comment (darcyYe) [inline src/sidebar/index.ts:34]: ditto

**Review by darcyYe — APPROVED** [review]
overall lgtm

File: src/sidebar/docs.ts (hunk 1 above)
Comment (charIeszhao) [reply src/sidebar/docs.ts:91]: It's deliberately commented out, giving an overview of the entire new structure

**Pull request merged**